onyo/commands/new.py

```diff
@@ -1,154 +1,296 @@
     """
-    Read a new word via user input for the creation of a asset name.
-    Checks for the serial field if it contains reserved characters, and checks
-    for all fields that they are not empty. If a check is not successful, it
-    tries to read the same word again.
+    Select the template to use. If `--template` is not given, use the template
+    from the repository config file.
 
-    Returns the word on success.
+    Returns the template path on success, or leaves with error.
     """
-    # read word for field from keyboard
-    word = input(word_description)
+    if template:
+        template = Path(repo.root, ".onyo/templates", template).resolve()
+    else:
+        conf = str(repo.get_config('onyo.new.template'))
+        template = Path(repo.root, ".onyo/templates", conf).resolve()
+    if not template.is_file():
+        print(f"Template '{template}' does not exist.", file=sys.stderr)
+        sys.exit(1)
+    return template
 
-    # if word contains reserved character, inform and read new word
-    for char in reserved_characters:
-        if word_description != "<serial>*:" and char in word:
-            log.info(f"'{char}' is in list of reserved characters: {reserved_characters}")
-            return read_new_word(word_description)
 
-    # if enter pressed without input, read new word
-    if len(word) == 0:
-        return read_new_word(word_description)
-    return word
+def valid_asset_destination(asset: Path, repo: Repo, existing_assets: set,
+                            new_assets: dict) -> None:
+    """
+    Test for an asset that its name is unique in the repository and input, that
+    the name follows the name scheme, and the directory is not protected.
+    """
+    if not repo.valid_name(asset):
+        print(f"'{asset}' is not a valid asset name.", file=sys.stderr)
+        sys.exit(1)
+    if file := [file for file in existing_assets if asset.name == file.name]:
+        print(f"Filename '{asset.name}' already exists as '{file[0]}'.", file=sys.stderr)
+        sys.exit(1)
+    elif file := [file for file in new_assets if asset.name == file.name]:
+        print(f"Input contains multiple '{file[0].name}'", file=sys.stderr)
+        sys.exit(1)
+    if repo._is_protected_path(asset):
+        print(f"The path is protected by onyo: '{asset}'", file=sys.stderr)
+        sys.exit(1)
 
 
-def run_onyo_new(directory, template, interactive, repo: Repo):
+def create_assets_in_destination(assets: dict, repo: Repo) -> None:
     """
-    Read a new asset name, check it for uniqueness and validity, create the
-    asset (from a template if one was given), edit the new asset (if not
-    suppressed with flag) and stage it with git.
-
-    Returns the newly created asset's filename on success.
+    Create assets and set their contents, and if their directory does not exist
+    yet, create it.
     """
+    for asset in assets.keys():
+        # create missing directories
+        if not asset.parent.is_dir():
```

Comment:
What happens in the pathological case where the "parent" is actually a file/asset?